tests/checkpointing/test_model_checkpoint.py

```diff
@@ -402,7 +402,6 @@ def test_model_checkpoint_no_extraneous_invocations(tmpdir):
         max_epochs=num_epochs,
     )
     trainer.fit(model)
-    assert trainer.state.finished, f"Training failed with {trainer.state}"
```

Comment:
Here is one blocker I am facing, in short: 

```python
trainer = Trainer(strategy="ddp_spawn", devices=2) 
trainer.fit(model)
assert trainer.state == TrainerState.finished
```

The main process spawns immediately 2 processes with the fit call (as proposed by this new design). These workers will update the state in their own instance of the trainer but once processes terminate/join, these changes will not be reflected in the main process, which has never seen any updates (as expected). 


A solution to this would be to put the trainer.state variable onto the queue and restore it in the main process. I am wondering if this is worth it. RFC from reviewers.

Reply:
Why don't we spawn N-1 processes and use the main proc as well?

Other then that we could probably send it through the queue as that should be quite cheap to do.